Source/JavaScriptCore/dfg/DFGObjectAllocationSinkingPhase.cpp

```diff
@@ -2058,6 +2207,8 @@ class ObjectAllocationSinkingPhase : public Phase {
                 bool desiredNextExitOK = node->origin.exitOK && !clobbersExitState(m_graph, node);
 
                 bool doLower = false;
+                m_nodeIndex = nodeIndex;
+                bool enableLengthMaterialization = true;
```

Comment:
Let's remove this as it is too complicated.

Reply:
We only need to materialize the constant (the replacment) for `GetArrayLength` once which is at this site. And `handleNode` is used in multiple places.